### PR TITLE
GuiHandle#setTextField(String, String) should call Gui methods on JavaFx threads #433

### LINKED ISSUE #433: GuiHandle#setTextField(String, String) should call Gui methods on JavaFx threads
On some computers, `GuiHandle#setTextField(String, String)` throws an error as follows:

```
Exception in thread "main" java.lang.IllegalStateException: This operation is permitted on the event thread only; currentThread = main
    at com.sun.glass.ui.Application.checkEventThread(Application.java:443)
    at com.sun.glass.ui.View.getNativeView(View.java:449)
    at com.sun.glass.ui.win.WinAccessible.get_HostRawElementProvider(WinAccessible.java:672)
    at com.sun.glass.ui.win.WinAccessible.UiaRaiseAutomationPropertyChangedEvent(Native Method)
    at com.sun.glass.ui.win.WinAccessible.sendNotification(WinAccessible.java:358)
    at javafx.scene.Node.notifyAccessibleAttributeChanged(Node.java:9602)
    at javafx.scene.control.TextInputControl$TextProperty.controlContentHasChanged(TextInputControl.java:1333)
    at javafx.scene.control.TextInputControl$TextProperty.access$1600(TextInputControl.java:1300)
    at javafx.scene.control.TextInputControl.lambda$new$162(TextInputControl.java:139)
    at com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(ExpressionHelper.java:137)
    at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:81)
    at javafx.scene.control.TextField$TextFieldContent.insert(TextField.java:87)
    at javafx.scene.control.TextInputControl.replaceText(TextInputControl.java:1204)
    at javafx.scene.control.TextInputControl.filterAndSet(TextInputControl.java:1171)
    at javafx.scene.control.TextInputControl.access$900(TextInputControl.java:79)
    at javafx.scene.control.TextInputControl$TextProperty.doSet(TextInputControl.java:1410)
    at javafx.scene.control.TextInputControl$TextProperty.set(TextInputControl.java:1323)
    at javafx.scene.control.TextInputControl.setText(TextInputControl.java:349)
    at guitests.guihandles.GuiHandle.setTextField(GuiHandle.java:65)
    at guitests.guihandles.CommandBoxHandle.enterCommand(CommandBoxHandle.java:29)
    at guitests.guihandles.CommandBoxHandle.runCommand(CommandBoxHandle.java:41)
    at guitests.AddCommandTest.assertAddSuccess(AddCommandTest.java:45)
    at guitests.AddCommandTest.add(AddCommandTest.java:27)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
    at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    at org.junit.runners.Suite.runChild(Suite.java:128)
    at org.junit.runners.Suite.runChild(Suite.java:27)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.j
```

This indicates that we are not permitted to call Gui methods unless we are on the JavaFx thread.

As it is not reproducible on all computers, the best we can do is to ensure that the appropriate methods called in `GuiHandle#setTextField(String, String)` really do run on the JavaFx thread.

### PR DESCRIPTION
**Pull request by yamgent**
Fixes #433

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@yamgent submitted v1 for review.

* [v1 1/1] [GuiHandle: Modify setTextField\(\) to run on JavaFx thread](https://github.com/se-edu/addressbook-level4/pull/434/commits/c89f9bd082b9e73b81b65f1aa84ea2fe03a2bfd6)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/cc22ed642b34f15798facec1bddd597aa8c0c395...c89f9bd082b9e73b81b65f1aa84ea2fe03a2bfd6))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/434/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: src/test/java/guitests/guihandles/GuiHandle.java [hunk 1]
```diff
@@ -62,7 +63,7 @@ protected String getTextFieldText(String filedName) {
     protected void setTextField(String textFieldId, String newText) {
         guiRobot.clickOn(textFieldId);
         TextField textField = getNode(textFieldId);
-        textField.setText(newText);
+        Platform.runLater(() -> textField.setText(newText));
```
Comment (pyokagan) [inline src/test/java/guitests/guihandles/GuiHandle.java]: The [`Platform.runLater` docs](https://docs.oracle.com/javase/8/javafx/api/javafx/application/Platform.html#runLater-java.lang.Runnable-) says that:

> Run the specified Runnable on the JavaFX Application Thread at some unspecified time in the future. 

The key being "unspecified time in the future", which means that `textField.setText()` may not actually run when this `setTextField()` method returns, which will break the expectation of callers.

GuiHandles should have access to the `FxRobot`? I think you could use its `interact()` method instead?

Comment (yamgent) [inline src/test/java/guitests/guihandles/GuiHandle.java]: Thanks for pointing that out, should have looked at the rest of the `GuiHandle.java` to see how they manage to accomplish it. :)

**Comment by CanIHasReview** [comment]
# v2

@yamgent submitted v2 for review.

* [v2 1/1] [GuiHandle: Modify setTextField\(\) to run on JavaFx Gui thread](https://github.com/se-edu/addressbook-level4/pull/434/commits/1aaaeae3cd32dc3dcae80e8916c17236e77af92e)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/6af393367c998910784881146ffa049725bc31bd...1aaaeae3cd32dc3dcae80e8916c17236e77af92e)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/80cd2426db168c1824f5a2623576500eaf7ab6c2~1...80cd2426db168c1824f5a2623576500eaf7ab6c2))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/434/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by yamgent** [comment]
@damithc Need you to reconfirm that the new fix works on your computer.

**Comment by damithc** [comment]
@yamgent tested again. Still working.

**Pull request merged**